### ISSUE TITLE
Questions about testing on GoPro dataset

### ISSUE THREAD

**Issue by pp00704831** [linjing7/VR-Baseline#12]
Hello, 

I follow your instruction and recollect  GoPro dataset, for example: ./GoPro/test/blur/GOPR0384_11_00/000000.png.
I run the test code by "python demo/restoration_video_demo.py  ./configs/FGST_deblur_gopro.py  FGST.pth  ./data/GoPro  results
However, an error occurs "No such file or directory: './data/GoPro/00000000.png'."

It seems like the loader does not load the name of the video files, what should I do to test successfully?

**Comment by linjing7** [comment]
Hi, I'm sorry that the `demo/restoration_video_demo.py` can only test a video each time. So you may need to change the video path to `./GoPro/test/blur/GOPR0384_11_00/xxx`. I'll fix this problem and train FGST on DVD dataset after CVPR ddl.

**Comment by pp00704831** [comment]
Thank you for your reply.  

I have a question about the inference time on the GoPro test set. How do you measure the inference time?
I measure the synchronized time in the class FGST(nn.Module) of FGST.py, where you would load 100 frames and predict 100 deblurred results.
I measure the inference time per frame in the forward function of FGST.py as

```
class FGST(nn.Module):
                 .
                 .
                 .
      def forward(self, x):
              """
              :param x: [n,t,c,h,w]
              :return: out: [n,t,c,h,w]
              """
              n, t, c, h_input, w_input = x.size()
      
              torch.Cuda.synchronize()
              start = time.time()
                           .
                           .
                           .
              torch.cuda.synchronize()
              stop = time.time()
              print(Time:, (stop - start) / t)
      
              return outputs[:, :, :, :h_input, :w_input]              
```
But the time is pretty large about 2.64 (s). How do you measure the inference time?

**Comment by linjing7** [comment]
Hi, @pp00704831 , thanks for your interest. We test the inference time during distributed training with 8 GPUs. In each epoch, the method would validate on the GOPRO test set and we obtain a validation time. And then, we divide the validation time by the frame number of GOPRO test set to get the inference time per frame.

**Comment by pp00704831** [comment]
Hi,

Thank you for your reply. BTW, could you share the GoPro deblurred results for us to download?

**Comment by linjing7** [comment]
Please refer to this [issue](https://github.com/linjing7/VR-Baseline/issues/11).

**Comment by pp00704831** [comment]
Hi,

I tested your model on the pre-trained weights, but the figure seems to have some artifacts. Do you have this problem before?
I did not change the code you gave and I test the results and get PSNR=31.54.
Do you have any idea about this? 

![image](https://user-images.githubusercontent.com/57279608/200842853-788158a0-f947-4acd-8c7a-70932f13b07f.png)

**Comment by linjing7** [comment]
Hi, the original pre-trained model can not be transferred outside due to company constraints.  The pre-trained model provided here is retrained in school and I have never visualized the results.  I will solve this problem and provide the deblurred results after CVPR ddl. 

It may be caused by the resolution difference during training and testing. You can try splitting the image into 256x256 patches and then test them seperately.

**Comment by pp00704831** [comment]
Hi, got it! Wish you all the best!

**Comment by linjing7** [comment]
@pp00704831 Hi, I have fixed the bug of testing code. You can update the code and test FGST on GoPro dataset with the following command:
`bash tools/dist_train.sh configs/FGST_deblur_gopro_test.py 8`

The evaluation result is 33.05 dB and I have not found the artifact you showed. BTW, I've provided the data preparation of DVD dataset and I am retraining FGST on DVD and GoPro.
